bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/discovery/ElroConnectsDiscoveryService.java

```diff
@@ -122,8 +122,8 @@ public void deactivate() {
     @Override
     public void setThingHandler(@Nullable ThingHandler handler) {
         ElroConnectsBridgeHandler bridge = null;
-        if (handler instanceof ElroConnectsBridgeHandler) {
-            bridge = (ElroConnectsBridgeHandler) handler;
+        if (handler instanceof ElroConnectsBridgeHandler bhandler) {
+            bridge = bhandler;
```

Comment:
This could be simplified further by removing line 124 and lines 130-132:
```suggestion
        if (handler instanceof ElroConnectsBridgeHandler bridgeHandler) {
            this.bridgeHandler = bridgeHandler;
            bridgeHandler.setDiscoveryService(this);
```

Reply:
`this.` should be removed from last line like in the suggested code to avoid null annotation issue (other thread could have set `this.bridgeHandler` to null in meantime).